src/ndarray/ndarray.cc

```diff
@@ -402,7 +402,7 @@ inline void CopyFromToCsrImpl(const NDArray& from, const NDArray& to, RunContext
   // if source storage is not initialized, fill destination with zeros
   auto s = ctx.get_stream<to_xpu>();
   if (!from.storage_initialized()) {
-    op::FillZerosCsrImpl<to_xpu>(s, to);
+    op::FillZerosCsrImpl(s, const_cast<NDArray *>(&to));
```

Comment:
const cast should be avoided

Reply:
I agree!  Why the code passes around const references to items that will be modified is a problem.  It's both counter-intuitive -- making the code harder to read and understanbd -- but it's also dangerous since the compiler optimizer may interpret the const in its value-casting strategy.